manim/utils/bezier.py

```diff
@@ -629,67 +1410,106 @@ def get_quadratic_approximation_of_cubic(
     T0 = h0 - a0
     T1 = a1 - h1
 
-    # Search for inflection points.  If none are found, use the
-    # midpoint as a cut point.
-    # Based on http://www.caffeineowl.com/graphics/2d/vectorial/cubic-inflexion.html
-    has_infl = np.ones(len(a0), dtype=bool)
-
+    # Search for inflection points. This happens when the acceleration (2nd derivative)
+    # is either zero or perpendicular to the velocity (1st derivative), captured
+    # in the cross product equation C'(t) x C''(t) = 0.
+    # If no inflection points are found, use the midpoint as the split point instead.
+    # Based on https://pomax.github.io/bezierinfo/#inflections
+    t_split = np.full(a0.shape[0], 0.5)
+
+    # Let C(t) be a cubic Bézier curve defined by [a0, h0, h1, a1].
+    # The below variables p, q, r allow for expressing the curve in a
+    # polynomial form convenient for derivatives:
+    # C(t) = a0 + 3tp + 3t²q + t³r
     p = h0 - a0
     q = h1 - 2 * h0 + a0
     r = a1 - 3 * h1 + 3 * h0 - a0
 
+    # Velocity:      C'(t) = 3p + 6tq + 3t²r
+    # Acceleration: C''(t) = 6q + 6tr
+    #       C'(t) x C''(t) = 18 (t²(qxr) + t(pxr) + (pxq)) = 0
+    # Define a = (qxr), b = (pxr) and c = (pxq).
+    # If C(t) is a 2D curve, then a, b and c are real numbers and
+    # this is a simple quadratic equation.
+    # However, if it's a 3D curve, then a, b and c are 3D vectors
+    # and this would require solving 3 quadratic equations.
+    # TODO: this simplifies by considering the 2D case. It might fail with 3D curves!
```

Comment:
We need the 3D part, remove Todo

Reply:
Discussed in Discord, but I clarify for the readers: the original algorithm already assumed a 2D case and used the `cross2D` functions. I was not the one to make that simplification, I just documented better what was already there in the case of `get_quadratic_approximation_of_cubic`.
Therefore, this comment was dismissed. :+1: